manager/src/main/java/com/redhat/service/bridge/manager/ProcessorServiceImpl.java

```diff
@@ -186,28 +181,22 @@ public Long getProcessorsCount(String bridgeId, String customerId) {
     }
 
     @Override
-    public void deleteProcessor(String bridgeId, String processorId, String customerId) {
-        Processor processor = doDeleteProcessor(bridgeId, processorId, customerId);
-
-        workManager.schedule(processor);
-    }
-
     @Transactional
-    protected Processor doDeleteProcessor(String bridgeId, String processorId, String customerId) {
+    public void deleteProcessor(String bridgeId, String processorId, String customerId) {
         Processor processor = processorDAO.findByIdBridgeIdAndCustomerId(processorId, bridgeId, customerId);
         if (processor == null) {
             throw new ItemNotFoundException(String.format("Processor with id '%s' does not exist on bridge '%s' for customer '%s'", processorId, bridgeId, customerId));
         }
-        processor.setStatus(ManagedResourceStatus.DEPROVISION);
+
+        // Processor and Connector deletion and related Work creation should always be in the same transaction
+        processorDAO.getEntityManager().merge(processor).setStatus(ManagedResourceStatus.DEPROVISION);
+        connectorService.deleteConnectorEntity(processor);
```

Comment:
Could you explain me a little bit the `ConnectorsService` naming. That interface seems to be tightly connected with `Processors`. Just asking, shouldn't be the names more like:
- `ProcessorsService`
- `createProcessor`
- `deleteProcessor`

Or is the naming as it is because, most Processors use some `Connectors` ?

Reply:
Hi @jomarko I doubt there is any convention about the naming of methods in the four services, `ShardService`, `BridgeService`, `ProcessorService` and `ConnectorService`. I suspect they've been written by different people at different times. I've merely come along to make some changes. There probably is room for consistency in naming but it's arguably out of the scope of this JIRA/PR. @lucamolteni @r00ta @robpblake Would you like to impose some consistency?

Reply:
@manstis yes, I approved the PR. For sure not related to it. Just surprised that `ConnectorsService` interface methods take as argument `processor`, when we have `ProcessorService` already.